dkg/state_machine.go

```diff
@@ -613,14 +613,14 @@ func ValidateProposal(currentState *DBState, terms *drand.ProposalTerms) error {
 		return err
 	}
 
-	if terms.TransitionTime == nil {
-		return ErrTransitionTimeMissing
-	}
-
 	if terms.Epoch == 1 {
 		return validateFirstEpoch(terms)
 	}
 
+	if terms.TransitionTime == nil {
+		return ErrTransitionTimeMissing
+	}
+
```

Comment:
Wait, what if `terms.TransitionTime == nil && terms.Epoch == 1` what's expected?

The code in this file is lacking quite a few comments, e.g. about what/why `validateFirstEpoch` is different for the first one.

Reply:
for the first epoch, the `TransitionTime` gets set to `GenesisTime`

Reply:
I can certainly add some more comments though